tests/javascript_tests/test_shape_controller.js

```diff
@@ -0,0 +1,145 @@
+requirejs([
+    "jquery",
+    "underscore",
+    "model",
+    "view",
+    "viewcontroller",
+    "three",
+    "shape-controller",
+    "shape-editor",
+    "shapes"
+], function($, _, model, DecompositionView, viewcontroller, THREE,
+            ShapeController, ShapeEditor, shapes) {
+  $(document).ready(function() {
+    var EmperorAttributeABC = viewcontroller.EmperorAttributeABC;
+    var DecompositionModel = model.DecompositionModel;
+
+    module("Shape Controller", {
+      setup: function() {
+        // setup function
+        shapesAvailable = ['sphere', 'cube', 'cone', 'icosahedron', 'cylinder'];
```

Comment:
Alternatively, just add to `this` so you can have access to it, this is the equivalent of `self` in python.